server/src/main/java/io/crate/replication/logical/MetadataTracker.java

```diff
@@ -201,36 +209,96 @@ static ClusterState updateMetadata(String subscriptionName, ClusterState subscri
                 }
             }
         }
-        // Check for all the subscribed tables if the index metadata changed and apply
+        // Check for all the subscribed tables if the index metadata and settings changed and if so apply
         // the changes from the publisher cluster state to the subscriber cluster state
-        var metadataBuilder = Metadata.builder(subscriberClusterState.metadata());
-        var isUpdated = false;
+        var updatedMetadataBuilder = Metadata.builder(subscriberClusterState.metadata());
+        var updateClusterState = false;
         for (var followedTable : subscribedTables) {
+            var isIndexMetadataUpdated = false;
             var publisherIndexMetadata = publisherClusterState.metadata().index(followedTable.indexNameOrAlias());
             var subscriberIndexMetadata = subscriberClusterState.metadata().index(followedTable.indexNameOrAlias());
             if (publisherIndexMetadata != null && subscriberIndexMetadata != null) {
-                var publisherMapping = publisherIndexMetadata.mapping();
-                var subscriberMapping = subscriberIndexMetadata.mapping();
-                if (publisherMapping != null && subscriberMapping != null) {
-                    if (publisherIndexMetadata.getMappingVersion() > subscriberIndexMetadata.getMappingVersion()) {
-                        var indexMetadataBuilder = IndexMetadata.builder(subscriberIndexMetadata).putMapping(
-                            publisherMapping).mappingVersion(publisherIndexMetadata.getMappingVersion());
-                        metadataBuilder.put(indexMetadataBuilder.build(), true);
-                        isUpdated = true;
-                    }
+                var updatedIndexMetadataBuilder = IndexMetadata.builder(subscriberIndexMetadata);
+                isIndexMetadataUpdated =
+                    updateIndexMetadataSettings(updatedIndexMetadataBuilder, publisherIndexMetadata, subscriberIndexMetadata, indexScopedSettings) ||
+                    updateIndexMetadataMapping(updatedIndexMetadataBuilder, publisherIndexMetadata, subscriberIndexMetadata);
```

Comment:
This will only update the settings if `updateIndexMetadataSettings` returns true.

```
jshell> Function<Boolean, Boolean> f = (b) -> { System.out.println(b); return b;};
f ==> $Lambda$21/0x0000000800c09a08@7291c18f

jshell> var res = f.apply(true) || f.apply(false);
true
res ==> true
```

Reply:
The reason for tracking the changes is to avoid unnecessary version updates. Therefore the index metadata should only be updated if there was a change at the settings or the mappings.

Reply:
Ah damn, also expressed myself wrongly. My point was that `updateIndexMetadataMapping()` will only be called if `updateIndexMetadataSettings()` returns `false`, see my code example.

Reply:
Great catch, thank you. I simplified the whole workflow.